src/models/db.js

```diff
@@ -28,6 +28,19 @@ const IDLE           = 0
 const INITIALIZING   = 1;
 const READY          = 2;
 
+const dbHooks = [
+  'beforeValidate',
+  'afterValidate',
+  'beforeCreate',
+  'afterCreate',
+  'beforeDestroy',
+  'afterDestroy',
+  'beforeRestore',
+  'afterRestore',
+  'beforeUpdate',
+  'afterUpdate'
+];
```

Comment:
I've started with the "instance" hooks because those seem most relevant to a client.